src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -252,6 +253,39 @@ public function configureOptions(OptionsResolver $resolver)
             return $options['required'] ? null : '';
         };
 
+        // BC closure, to be removed in 3.0
+        $choicesNormalizer = function (Options $options, $choices) use (&$choiceLabels) {
+            // Unset labels from previous invocations
+            $choiceLabels = array();
+
+            // This closure is irrelevant when "choices_as_values" is set to true
+            if ($options['choices_as_values']) {
+                return $choices;
+            }
+
+            ChoiceType::normalizeLegacyChoices($choices, $choiceLabels);
+
+            return $choices;
+        };
+
+        // BC closure, to be removed in 3.0
+        $choiceLabel = function (Options $options) use (&$choiceLabels) {
+            // If the choices contain duplicate labels, the normalizer of the
+            // "choices" option stores them in the $choiceLabels variable
+
+            // Trigger the normalizer
+            $options->offsetGet('choices');
+
+            // Pick labels from $choiceLabels if available
+            if (count($choiceLabels) > 0) {
+                return function ($choice, $key) use ($choiceLabels) {
```

Comment:
This is without a `&` by intention. The outer `$choiceLabels` lives in the type's scope, i.e. every field uses the same variable to transfer information from `choices` to `choice_label` during its construction. The inner closure, however, lives in the scope of the field and is used to generate ChoiceView instances during view construction. If we'd use the global variable instead, all choice fields in a form would have the labels of the last one, which is not what we want.